src/debugAdapter/goDebug.ts

```diff
@@ -479,71 +487,78 @@ class Delve {
 		});
 	}
 
-	callPromise<T>(command: string, args: any[]): Thenable<T> {
+	public callPromise<T>(command: string, args: any[]): Thenable<T> {
 		return new Promise<T>((resolve, reject) => {
 			this.connection.then(conn => {
-				conn.call<T>('RPCServer.' + command, args, (err, res) => {
-					if (err) return reject(err);
-					resolve(res);
+				conn.call<T>(`RPCServer.${command}`, args, (err, res) => {
+					return err ? reject(err) : resolve(res);
 				});
 			}, err => {
 				reject(err);
 			});
 		});
 	}
 
-	close(): Thenable<void> {
+	/**
+	 * To close the debugging session we do the following; If its a local process,
+	 * we try to halt and detach within a timeout otherwise we kill the debugger. If its a remote
+	 * process we try to halt and restart the remote target process running under the debugger.
```

Comment:
@vladbarosan For remote debugging, we don't seem to restart the remote delve process anymore. I dont' recall the discussion that lead to this decision.

I remember saying that we can consider using detach without the kill option instead of restart, but the current code calls detach with the kill option.

Reply:
@ramya-rao-a  yes I remember, let me test the behavior , and ill update

Reply:
@vladbarosan https://github.com/Microsoft/vscode-go/issues/1609#issuecomment-410111323 is another reference to the need for detach (without kill) instead of restart in case of remote debugging